client/src/formatter/forgeFormatter.ts

```diff
@@ -18,10 +20,8 @@ export async function formatDocument(
   const forgeCommand = await resolveForgeCommand();
 
   const formatted = await new Promise<string>((resolve, reject) => {
-    const currentDocument = vscode.window.activeTextEditor?.document.uri;
-    const rootPath = currentDocument
-      ? vscode.workspace.getWorkspaceFolder(currentDocument)?.uri.fsPath
-      : undefined;
+    const documentUri = vscode.window.activeTextEditor?.document.uri.fsPath;
+    const rootPath = documentUri && path.dirname(documentUri);
```

Comment:
Can you walk me through this change. We are switching from running forge in the workspace folder (which fails if your a monorepo and the foundry project is a subfolder).

Instead we run forge fmt in the directory of the file to be formatted and forge does the work of finding the relevant `foundry.toml`?